recipes/lzham/all/conanfile.py

```diff
@@ -0,0 +1,221 @@
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import (
+    apply_conandata_patches,
+    copy,
+    export_conandata_patches,
+    get,
+    replace_in_file,
+    rmdir
+)
+from conan.tools.microsoft import (
+    MSBuild, MSBuildDeps, MSBuildToolchain, VCVars, is_msvc, vs_layout
+)
+
+required_conan_version = ">=1.52.0"
+
+SLN_FILE = "lzham.sln"
+
+
+class PackageConan(ConanFile):
+    name = "lzham"
+
+    description = (
+        "Compression algorithm similar compression ratio and faster "
+        "decompression than LZMA."
+    )
+
+    license = "LicenseRef-LICENSE"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/richgel999/lzham_codec"
+    topics = ("compression", "lz-compression")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+        if not is_msvc(self):
+            # Remove lzhamtest from root CMakeLists.txt.
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, "CMakeLists.txt"),
+                "add_subdirectory(lzhamtest)\n",
+                ""
+            )
+
+            # This line in the root CMakeLists.txt can cause issues, see
+            # https://cmake.org/cmake/help/latest/policy/CMP0077.html.
+            replace_in_file(
+                self,
+                os.path.join(self.source_folder, "CMakeLists.txt"),
+                "option(BUILD_SHARED_LIBS \"build shared/static libs\" ON)",
+                ""
+            )
```

Comment:
https://github.com/conan-io/conan-center-index/blob/41adf57553fa3ab9088462d390a2c444fdc7f052/recipes/re2/all/conanfile.py#L50-L51 would be the prefered way